wifiphisher/common/macmatcher.py

```diff
@@ -1,133 +1,61 @@
+    def __init__(self, vendor_file):
+        # type: (Text) -> None
+        """Initialize the class with all the given arguments."""
+        self.logger = getLogger(__name__)
+        self._vendors = dict()  # type: Dict[Text, Text]
+        self._vendor_file = vendor_file
 
         # get the information in the vendor file
         self._get_vendor_information()
 
     def _get_vendor_information(self):
-        """
-        Read and process all the data in the vendor file
-
-        :param self: A MACMatcher object
-        :type self: MACMatcher
-        :return: None
-        :rtype: None
-        """
-
-        # open the file with all the MAC addresses and
-        # vendor information
-        with open(self._vendor_file, 'r') as _file:
-            # check every line in the file
+        # type: () -> None
+        """Read and process all the data in the vendor file."""
+        self.logger.debug("Parsing the vendor file")
+        with open(self._vendor_file) as _file:
             for line in _file:
                 # skip comment lines
                 if not line.startswith("#"):
-                    # separate vendor and MAC addresses and add it
-                    # to the dictionary
-                    separated_line = line.rstrip('\n').split('|')
-                    mac_identifier = separated_line[0]
-                    vendor = separated_line[1]
-                    logo = separated_line[2]
-                    self._mac_to_vendor[mac_identifier] = (vendor, logo)
+                    vendor_id, vendor_name = line.rstrip('\n').split("\t")
+                    self._vendors[vendor_id] = vendor_name
+                    self.logger.debug(
+                        "{vendor_name} with ID {vendor_id} has been added to database".
+                        format(vendor_name=vendor_name, vendor_id=vendor_id))
 
     def get_vendor_name(self, mac_address):
-        """
-        Return the matched vendor name for the given MAC address
-        or Unknown if no match is found
-
-        :param self: A MACMatcher object
-        :param mac_address: MAC address of device
-        :type self: MACMatcher
-        :type mac_address: string
-        :return: The vendor name of the device if MAC address is found
-                 and Unknown otherwise
-        :rtype: string
-        """
-
-        # Don't bother if there's no MAC
-        if mac_address is None:
-            return None
-
-        # convert mac address to same format as file
-        # ex. 12:34:56:78:90:AB --> 123456
-        mac_identifier = mac_address.replace(':', '').upper()[0:6]
-
-        # try to find the vendor and if not found return unknown
-        try:
-            vendor = self._mac_to_vendor[mac_identifier][0]
-            return vendor
-        except KeyError:
-            return "Unknown"
+        # type: (Text) -> Text
+        """Return the matched vendor name for the given MAC address.
 
-    def get_vendor_logo_path(self, mac_address):
```

Comment:
We are not allowed to put logos in the repo but that doesn't mean we can't support the feature, right? A user may get permission to use the copyrighted material.